src/utils/gasles_fee_calculator.ts

```diff
@@ -0,0 +1,200 @@
+import { BigNumber } from '@0x/utils';
+import { ZERO } from '../constants';
+import {
+    GasFee,
+    GasFeeConfig,
+    GaslessFeeConfigs,
+    GaslessFees,
+    IntegratorFee,
+    IntegratorFeeConfig,
+    ZeroexFee,
+    ZeroExFeeConfig,
+} from '../types';
+
+/**
+ * `transferFrom` estimated gas:
+ * - Decrease balance of the owner (SLOAD + SSTORE): 24,000
+ * - Increase balance of the spender (SLOAD + SSTORE): 24,000
+ * - Update allowance of the spender (SLOAD + SSTORE): 24,000
+ */
+const TRANSFER_FROM_GAS = new BigNumber(72e3);
+
+/**
+ * Calculate gasless fees object which contains total fee amount and a breakdown of integrator, 0x and gas fees.
+ *
+ * @param feeConfigs Gasless fee configs parsed from input.
+ * @param opts sellToken: Address of the sell token.
+ *             sellTokenAmount: Amount of the sell token.
+ *             sellTokenAmountPerBaseUnitNativeToken: Amount of sell token per base unit native token.
+ *             gasPrice: Estimated gas price.
+ *             quoteGasEstimate: The gas estimate to fill the quote.
+ * @returns Gasless fees object and the total fee amount.
+ */
+export function calculateGaslessFees(opts: {
```

Comment:
I wonder if we need to call this `calculateGaslessFees` or if we could just call this `calculateFees` and if the gas potion of the `feeConfigs` is missing, we just don't calculate gas. 

So in other words, we generalize this class to `fee_calculator` - and it works in both a gasless and regular setting

Reply:
otherwise, I think this looks great!